Storage/ThirdParty/SwiftData.swift

```diff
@@ -179,13 +172,17 @@ open class SwiftData {
                 deferred.running = false
             }
 
+            if !self.closed && self.sharedConnection == nil {
```

Comment:
Double check that this work happens on the same queue on which `self.closed` is touched. I see `forceClose` does it right, but is there anywhere else?

Reply:
Checking... valid concern there...